### ISSUE TITLE
`padding-line-between-statements`: newline-before-first-import (JSDoc)

### ISSUE THREAD

**Issue by Deadmano** [eslint-stylistic/eslint-stylistic#654]
### Clear and concise description of the problem

I would like to be able to add a newline above the first `import` statement in a file if there is no newline between a JSDoc comment (usually a license header) and the first import.

### Suggested solution

Consider adding a way to specify a `JSDoc` block to the existing [`padding-line-between-statements`](https://eslint.style/rules/js/padding-line-between-statements) rule, that way `prev` could be of a `STATEMENT_TYPE` `jsdoc` that would detect the JSDoc pattern.

### Additional context

**Invalid:**
```JS
/**
 * @license MIT
 * This file is part of Example Project.
 * Copyright © Author
 */
import { CommentState, FormattingState, LineState } from '../config/types';
```
---
**Valid:**
```JS
/**
 * @license MIT
 * This file is part of Example Project.
 * Copyright © Author
 */

import { CommentState, FormattingState, LineState } from '../config/types';
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by Deadmano** [comment]
@9romise I believe this should be re-opened as it is a unique enough issue since the one you marked this is a duplicate of would not resolve this issue, unless the alternative approach was taken up and it still would not be an official implementation but rather a user-created one.

**Comment by 9romise** [comment]
From what I understand, adding a `comment` could also solve this issue since JSDoc counts as comments. Are there other considerations?

**Comment by Deadmano** [comment]
Could you go into a bit more detail about what you mean or what I might be missing from the linked issue? As I understand it, the suggested solution, or rather the alternative suggestion, would provide a custom way to specify matching blocks via configuration. What I'm requesting is native JSDoc support that detects the format it uses as per the example I shared so that the existing rules would apply to it.

**Comment by 9romise** [comment]
I see two proposed solutions in #417:

1. Add `comment` detection - I think this could resolve this issue.
2. Introduce `custom-blocks` - Would enhance the rule

But I'm currently unsure whether this belongs in `padding-line-between-statements` or `lines-around-comment`.

**Comment by Deadmano** [comment]
I see what you mean now! Yes, especially with the custom blocks I feel this would be an acceptable solution, though it does still require manual configuration as opposed to built-in JSDoc support which is what I was asking for in this issue.
